WalletWasabi.Tests/UnitTests/WabiSabi/Client/WabiSabiApiClientTests.cs

```diff
@@ -0,0 +1,48 @@
+using Moq;
+using NBitcoin;
+using System.Threading.Tasks;
+using WalletWasabi.BitcoinCore.Rpc;
+using WalletWasabi.Crypto;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Tests.Helpers;
+using WalletWasabi.WabiSabi.Backend;
+using WalletWasabi.WabiSabi.Backend.Banning;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using WalletWasabi.WabiSabi.Client;
+using WalletWasabi.WabiSabi.Crypto;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.WabiSabi.Client
+{
+	public class WabiSabiApiClientTests
+	{
+		[Fact]
+		public async Task RegisterInputAsyncTest()
+		{
+			var config = new WabiSabiConfig();
+			var round = WabiSabiFactory.CreateRound(config);
+			using Arena arena = await WabiSabiFactory.CreateAndStartArenaAsync(config, round);
+
+			using var key = new Key(); 
+			var outpoint = BitcoinFactory.CreateOutPoint();
+
+			var mockRpc = new Mock<IRPCClient>();
+			mockRpc.Setup(rpc => rpc.GetTxOutAsync(outpoint.Hash, (int)outpoint.N, true))
+				.ReturnsAsync(new NBitcoin.RPC.GetTxOutResponse{
+					IsCoinBase = false,
+					Confirmations = 200,
+					TxOut = new TxOut(Money.Coins(1m), key.PubKey.WitHash.GetAddress(Network.Main)),
+				});
+			await using var coordinator = new PostRequestHandler(config, new Prison(), arena, mockRpc.Object);
```

Comment:
We will likely be repeating this kind of boilerplate a lot in the near future. I don't have concrete ideas right now, but we should keep that in mind and eventually come up with something simple (following 80/20 rule).

Some incomplete thoughts on what we will probably need in the long run:

1. multiple clients with multiple UTXOs
2. ability to support a sequence of several rounds, both blame rounds and remixing
3. specific UTXO allocations for those clients to test various edge cases for amount organization
4. a balance of defaults & control over round parameters so that simple cases don't have lots of boilerplate but we can still use this to obtain good coverage

Reply:
based on the other comment thread i think we can resolve this and bring it up in some later PR